src/pages/FoodItem/FoodItemPage.tsx

```diff
@@ -34,7 +34,30 @@ export default function FoodItemPage() {
     document.title = foodItem ? foodItem.name : "Not Found";
   });
 
-  const [totalPrice, setTotalPrice] = useState(foodItem.basePrice);
+  //Create price state. Updates on every input change.
+  const initialPrices = {
+    optionPrices: foodItem.options.map((option) => 0),
+    totalPrice: foodItem.basePrice,
+  };
+  const priceReducer = (
+    state: {
+      optionPrices: number[];
+      totalPrice: number;
+    },
+    action: {
+      index: number;
+      amount: number;
+    }
+  ) => {
+    const newState = { ...state };
```

Comment:
instead of using useReducer you can use a handler function instead

Reply:
useReducer is for multipler ways to update the state. Here there's only 1 https://dmitripavlutin.com/react-usereducer/

Reply:
There was an issue when I used a simple
`const [optionPrices, setOptionPrices] = useState([] as number[]);`

In subOption.tsx, I would update this variable whenever an checkbox/radio is clicked. So I use useEffect like this:
```
useEffect(() => {

    //get new option price based on subOptionFormValues
    let newOptionPrice = 0;
    subOptionFormValues.forEach((subOptionName) => {
      optionPrice +=
        option.subOptions.find((subOption) => subOption.name === subOptionName)
          ?.price ?? 0;
    });
    
    //Get and update the copy
    const optionPricesCopy = {...optionPrices};
    optionPricesCopy[optionIndex] = newOptionPrice;

    //Update state
    setOptionPrices(optionPricesCopy);

},[subOptionFormValues, optionPrices, setOptionPrices]);
```

This doesn't work, because I use optionPrices as a dependency, but it also sets optionPrices, so it will call itself infinitely. I need optionPrices as a dependency, because I get data from it by creating a copy of it with `optionPricesCopy = {...optionPrices}`, then I only change the value at the current optionIndex.

To fix this I use useReducer, then I can simply dispatch by passing in both index and payload. The state updating function is defined elsewhere, which has information of the previous state. So it will use index and payload to update the optionPrice array, and the final price. 

I can't think of a simpler way of doing this. Because at some point, I need an array of numbers in the parent component. This array has to be updated at some point. And when it is updated, it is updated **on each element**, not as an entire array. When I update an element inside an array and not the entire array, I will require a copy of it, the previous state. So I need useReducer.

Can you think of a better solution?

Reply:
Replaced useReducer with useState. Everything works as intended.